tests/reference/c-expr17-a800575.stdout

```diff
@@ -0,0 +1,54 @@
+#include <inttypes.h>
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <lfortran_intrinsics.h>
+
+#define ASSERT(cond)                                                           \
+    {                                                                          \
+        if (!(cond)) {                                                         \
+            printf("%s%s", "ASSERT failed: ", __FILE__);                       \
+            printf("%s%s", "\nfunction ", __func__);                           \
+            printf("%s%d%s", "(), line number ", __LINE__, " at \n");          \
+            printf("%s%s", #cond, "\n");                                       \
+            exit(1);                                                           \
+        }                                                                      \
+    }
+#define ASSERT_MSG(cond, msg)                                                  \
+    {                                                                          \
+        if (!(cond)) {                                                         \
+            printf("%s%s", "ASSERT failed: ", __FILE__);                       \
+            printf("%s%s", "\nfunction ", __func__);                           \
+            printf("%s%d%s", "(), line number ", __LINE__, " at \n");          \
+            printf("%s%s", #cond, "\n");                                       \
+            printf("%s", "ERROR MESSAGE:\n");                                  \
+            printf("%s%s", msg, "\n");                                         \
+            exit(1);                                                           \
+        }                                                                      \
+    }
+
+// Forward declarations
+void _lpython_main_program();
+void f();
+
+// Implementations
+void _lpython_main_program()
+{
+    f();
+}
+
+void f()
+{
+    bool b;
+    int32_t i;
+    i = 3;
+    b = (bool)(i);
```

Comment:
I don't think `C` has `bool` type. Probably, doing https://github.com/lcompilers/lpython/pull/717/files#r911632396 will bring clarity since the generated code will be compiled by `GCC` and `Clang` on CI.

Reply:
> I don't think `C` has `bool` type

If you see the C code that is generated, you'd see `stdbool.h` as a header file, so it should compile.  But I can shift this test under `integration_tests`,  just in case...

Reply:
Well checking if C code compiles is the best approach I feel. We have the infrastructure for that so why not use it.

Reply:
Try running `./run_tests.sh` under `integration_tests` directory.